test/parallel/test-http-agent-destroyed-socket.js

```diff
@@ -48,7 +48,7 @@ const server = http.createServer(common.mustCall((req, res) => {
     response.on('end', common.mustCall(() => {
       request1.socket.destroy();
 
-      response.socket.once('close', common.mustCall(() => {
+      request1.socket.once('close', common.mustCall(() => {
```

Comment:
This looks like it's no longer testing what it used to be testing (just working around the fact that `response.socket` is cleared now by attaching the event to an unrelated socket?).  Proper fix would be to leave the event on `response.socket` and just attach it before the socket is cleared (up 3 lines, right after `response.pipe`)?

I'm not actually familiar with this code, just in here since this test started breaking on Node v14 over in [http-parser-js](https://github.com/creationix/http-parser-js/blob/master/tests/parallel/test-http-agent-destroyed-socket.js), which has a fork of some of these tests, but I noticed the old code also had a big comment about how important the timing/ordering of attaching that event listener was.